### PR TITLE
[BB-233] shell section + terraform updates

### PR DESCRIPTION
**Pull request by Franr**
Document new `shell` command.

Update `terraform` doc showing mount/env-vars support.

### CODE REVIEW

File: docs/user-guide/leverage-cli/reference/shell.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  injection via mfa/sso.
+
+  Syntax: leverage shell --mount <local-path> <container-path> --env-var <name> <value>
+  Example: leverage shell --mount /home/user/bin/ /usr/bin/ --env-var env dev
+
+  Both mount and env-var parameters can be provided multiple times.
+  Example: leverage shell --mount /home/user/bin/ /usr/bin/ --mount /etc/config.ini /etc/config.ini --env-var init 5 --env-var env dev
+
+Options:
+  --mount <TEXT TEXT>...
```
Comment (diego-ojeda-binbash) [inline docs/user-guide/leverage-cli/reference/shell.md:23]:
```suggestion
  --mount <HOST_PATH> <CONTAINER_PATH>...
```
Is the above supported? Does it need to use quotes?

File: docs/user-guide/leverage-cli/reference/shell.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  --env-var <TEXT TEXT>...
```
Comment (diego-ojeda-binbash) [inline docs/user-guide/leverage-cli/reference/shell.md:24]:
```suggestion
  --env-var <NAME> <VALUE>...
```
Is the above supported? Or does it actually need to be like this: `--env-var=FOO=BAR`? Or even use quotes?

File: docs/user-guide/leverage-cli/reference/shell.md (hunk 1 above)
Comment (Franr) [reply docs/user-guide/leverage-cli/reference/shell.md:23]: no quotes, is the way Click show tuples unfortunately

We can change the documentation but `leverage shell --help` will still show it that way.
I think the clarification here: https://github.com/binbashar/le-ref-architecture-doc/pull/205/files#diff-de2efff7f5cb5f149039d444f61b11b87794aca85cd4664f5cfbcf582f180ac0R16
and the example below might be enough to clarify

File: docs/user-guide/leverage-cli/reference/shell.md (hunk 2 above)
Comment (Franr) [reply docs/user-guide/leverage-cli/reference/shell.md:24]: same than above

File: docs/user-guide/leverage-cli/reference/shell.md (hunk 1 above)
Comment (diego-ojeda-binbash) [reply docs/user-guide/leverage-cli/reference/shell.md:23]: That's ok. It looked strange to me to that it shows "<TEXT TEXT>" but the examples right above show clearly how to use the command.

**Review by juanmatias — APPROVED** [review]

**Pull request merged**